derrida/interventions/management/commands/annotation_data.py

```diff
@@ -68,13 +67,39 @@ def handle(self, *args, **kwargs):
             for intervention in data:
                 csvwriter.writerow(self.flatten_dict(intervention))
 
-    def intervention_data(self, intervention):
+    def annotation_data(self, intervention):
         '''Generate a dictionary of data to export for a single
         :class:`~derrida.books.models.Reference` object'''
 
         # NOTE: using OrderedDict to ensure JSON output follows logical
         # order in Python < 3.6, where dict order is not guaranteed
 
+        if intervention.canvas:
+            canvas_url = reverse('books:canvas-image', kwargs={
+                'slug': intervention.work_instance.slug,
+                'short_id': intervention.canvas.short_id, 
+                'mode': 'large',
+            })
+            iiif_base = ('https://derridas-margins.princeton.edu' + canvas_url).replace('/large/', '/iiif/')
+            page_iiif = iiif_base + 'full/500,/0/default.jpg'
+        else:
+            page_iiif = ''
+
```

Comment:
More of this is hard-coded and based on string replacement than I'd like (not very semantic/readable).

How about this approach?  (It's cheating a little using "images" as the canvas detail image id so we can generate base url as an "endpoint" that the piffle image client will understand)

```suggestion
      from derrida.common.utils import absolutize_url

      page_iiif = annotation_region = None
     
        if intervention.canvas:
           # get the piffle image api client for the canvas
           img = intervention.canvas.image
           # replace original iiif api endpoint and image id with local proxy iiif url
           img.api_endpoint = absolutize_url(reverse('books:canvas-detail', kwargs={'slug': intervention.work_instance.slug, 'short_id': 'images'})).rstrip('/')
           img.image_id = '%s/iiif' % i.canvas.short_id
           # request 500 width image and convert to str for iiif image url
           page_iiif = str(img.size(width=500))

```

Reply:
✅ 49244c4